### PR TITLE
Add vagrant configuration for testing purposes

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

I wrote simple vagrant scripts to have repeatable development environment 

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I definitely going to write README for vagrant if we would like to go with it for master. For now:

please create `vagrant/envs.sh` with your access and id values:

```
#!/bin/bash

ENDPOINT='sumologic api endpoint'
ACCESS_KEY='your sumologic access key'
ACCESS_ID='your sumologic access id'
```

File: vagrant/provision.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+microk8s.status --wait-ready
+ufw allow in on cbr0
+ufw allow out on cbr0
+ufw default allow routed
+
+microk8s.enable dashboard
+microk8s.enable registry
+microk8s.enable storage
+microk8s.enable dns
+microk8s.enable prometheus
```
Comment (perk-sumo) [inline vagrant/provision.sh]: This should go.

File: vagrant/provision.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+echo "--allow-privileged=true" >> /var/snap/microk8s/current/args/kube-apiserver
+systemctl restart snap.microk8s.daemon-kubelet.service
+systemctl restart snap.microk8s.daemon-apiserver.service
+
+# allow connections to outside
+iptables -P FORWARD ACCEPT
+apt-get install --yes iptables-persistent
+
+#snap alias microk8s.helm helm
+helm init --wait
```
Comment (perk-sumo) [inline vagrant/provision.sh]: This will work only for the root user. Please run as a `vagrant` user.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure('2') do |config|
+  config.vm.box = 'ubuntu/bionic64'
+  config.vm.box_check_update = false
+  config.vm.host_name = 'sumologic-kubernetes-collection-dev.vagrantup.com'
+
+  config.vm.network :forwarded_port, guest: 18080, host: 18080
```
Comment (perk-sumo) [inline Vagrantfile]: Port forwards are not needed with `:private_network`, those will just work out of the box. Please remove.

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure('2') do |config|
+  config.vm.box = 'ubuntu/bionic64'
+  config.vm.box_check_update = false
+  config.vm.host_name = 'sumologic-kubernetes-collection-dev.vagrantup.com'
```
Comment (perk-sumo) [inline Vagrantfile]: Please remove `-dev.vagrantup.com`

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  config.vm.network :forwarded_port, guest: 51820, host: 51820
+  config.vm.network :forwarded_port, guest: 8443, host: 8443
+
+  config.vm.network :private_network, ip: "192.168.33.33"
+
+  config.vm.provider 'virtualbox' do |vb|
+    vb.gui = false
+    vb.cpus = 8
+    vb.memory = 16384
+    vb.name = 'sumologic-kubernetes-collection-dev'
```
Comment (perk-sumo) [inline Vagrantfile]: Please remove `-dev`.

File: vagrant/provision.sh (hunk 2 above)
Comment (sumo-drosiek) [reply vagrant/provision.sh]: it works fine for me, why? 🤔

Comment (sumo-drosiek) [reply vagrant/provision.sh]: On the other hand
```
    default: + su vagrant -c helm init --wait
    default: bash: helm: command not found
    default: + usermod -a -G microk8s vagrant
    default: usermod: group 'microk8s' does not exist
```

Comment (perk-sumo) [reply vagrant/provision.sh]: AFAIK provisioning is run under root account.
As for those errors it seems microk8s has not been installed correctly.

File: vagrant/provision.sh (hunk 1 above)
Comment (perk-sumo) [reply vagrant/provision.sh]: I meant this line only: `microk8s.enable prometheus`

**Pull request merged**